src/parquet/column/scanner-test.cc

```diff
@@ -0,0 +1,271 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include <gtest/gtest.h>
+
+#include <algorithm>
+#include <cstdint>
+#include <cstdlib>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "parquet/types.h"
+#include "parquet/column/page.h"
+#include "parquet/column/scanner.h"
+#include "parquet/column/test-util.h"
+#include "parquet/schema/descriptor.h"
+#include "parquet/schema/types.h"
+#include "parquet/util/test-common.h"
+
+using std::string;
+using std::vector;
+using std::shared_ptr;
+
+namespace parquet_cpp {
+
+using schema::NodePtr;
+
+bool operator==(const Int96& a, const Int96& b) {
+  return a.value[0] == b.value[0] &&
+    a.value[1] == b.value[1] &&
+    a.value[2] == b.value[2];
+}
+
+bool operator==(const ByteArray& a, const ByteArray& b) {
+  return a.len == b.len && 0 == memcmp(a.ptr, b.ptr, a.len);
+}
+
+static int FLBA_LENGTH = 12;
+bool operator==(const FixedLenByteArray& a, const FixedLenByteArray& b) {
+  return 0 == memcmp(a.ptr, b.ptr, FLBA_LENGTH);
+}
+
+namespace test {
+
+template <int TYPE, typename T>
```

Comment:
`typename T` adds extra complexity here because you can use `type_traits<TYPE>::value_type`